packages/plugins/src/SharedConfigPlugin.ts

```diff
@@ -113,24 +127,97 @@ const SharedConfigPlugin: PluginType<SharedConfigPluginPage, SharedConfigPluginO
     }
 
     // apply closest shared config
-    let closestSharedConfigIndex = 0;
-    for (const pagePath of indexPagesWithoutConfig) {
-      for (let i = 0; i < sharedConfigFiles.length; i++) {
-        if (isWithin(sharedConfigFiles[i], pagePath)) {
-          closestSharedConfigIndex = i;
+    if (sharedConfigFiles.length > 0) {
+      let closestSharedConfigIndex = 0;
+      for (const pagePath of indexPagesWithoutConfig) {
+        for (let i = 0; i < sharedConfigFiles.length; i++) {
+          if (isWithin(sharedConfigFiles[i], pagePath)) {
+            closestSharedConfigIndex = i;
+          }
         }
+
+        const sharedConfigFile = path.posix.join(
+          path.posix.dirname(String(pagePath)),
+          options.filename
+        );
+
+        const closestSharedConfig = path.posix.resolve(
+          path.dirname(String(pagePath)),
+          sharedConfigFiles[closestSharedConfigIndex]
+        );
+        config.setAliases(closestSharedConfig, [sharedConfigFile]);
       }
+    }
+  },
+  async afterUpdate(mutableFilesystem, { sharedFilesystem, globalConfig, namespace }, options) {
+    const { applyNamespaceSharedConfig } = globalConfig.data;
 
-      const sharedConfigFile = path.posix.join(
-        path.posix.dirname(String(pagePath)),
-        options.filename
-      );
+    if (applyNamespaceSharedConfig === undefined) {
+      // there is no source that exists that has told us it needs to share a namespace shared-config
+      return;
+    }
 
-      const closestSharedConfig = path.posix.resolve(
-        path.dirname(String(pagePath)),
-        sharedConfigFiles[closestSharedConfigIndex]
-      );
-      config.setAliases(closestSharedConfig, [sharedConfigFile]);
+    // find all the entries that match the namespace the plugin is running against
+    const namespaceSharedConfigs: {
+      paths: string[];
+      rootPath: string;
+      namespace: string;
+    }[] = Object.keys(applyNamespaceSharedConfig)
+      .filter(key => {
+        const keyNamespace = key.split('~~')?.[0];
+        return keyNamespace === namespace;
+      })
+      .map(key => applyNamespaceSharedConfig?.[key] || []);
+
+    for (const namespaceSharedConfig of namespaceSharedConfigs) {
+      if (await mutableFilesystem.promises.exists(namespaceSharedConfig.rootPath)) {
```

Comment:
This is to guard against the actual write of the shared config.json that is a copy.  It might trigger afterUpdate again but not 100% sure